hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstant.java

```diff
@@ -46,14 +55,35 @@ public class HoodieInstant implements Serializable, Comparable<HoodieInstant> {
   public static final Comparator<HoodieInstant> COMPARATOR = Comparator.comparing(HoodieInstant::getTimestamp)
       .thenComparing(ACTION_COMPARATOR).thenComparing(HoodieInstant::getState);
 
+  public static final Comparator<HoodieInstant> STATE_TRANSITION_COMPARATOR =
+      Comparator.comparing(HoodieInstant::getStateTransitionTime)
```

Comment:
We can just use `COMPARATOR` here.

Reply:
I mean for `STATE_TRANSITION_COMPARATOR ` construction, `COMPARATOR` can be reused to save the redundant code.